Source/WebCore/platform/graphics/mac/controls/SwitchMacUtilities.mm

```diff
@@ -82,6 +100,28 @@ static float easeInOut(const float progress)
     return -2.0f * pow(progress, 3.0f) + 3.0f * pow(progress, 2.0f);
 }
 
+static FloatRect logicalRect(const FloatRect& rect, bool isVertical)
+{
+    auto logicalRect = rect;
+    if (isVertical)
+        logicalRect.setSize(logicalRect.size().transposedSize());
+    return logicalRect;
+}
+
+static FloatRect trackRectForBounds(const FloatRect& bounds, const FloatSize& size)
+{
+    auto offsetY = std::max(((bounds.height() - size.height()) / 2.0f), 0.0f);
+    return { FloatPoint { bounds.x(), bounds.y() + offsetY }, size };
+}
+
+static void rotateContextForVerticalWritingMode(GraphicsContext& context, const FloatPoint origin, float x, float y)
```

Comment:
The `x` parameter here is unnecessary, since it's always going to be the height. And the `y` parameter is specific to a single adjustment (on/off) labels, that does not need to be generalized.

This could just be `rotateContextForLogicalRect(GraphicsContext& context, FloatRect& rect)`. And any additional translation you need (like the on/off) label can be made inline.

Eventually, this could just be a method somewhere like `ControlMac`, so `MeterMac` and `ProgressBarMac` could also use it. That would be for another PR though.